--- a/fmriprep/config.py
+++ b/fmriprep/config.py
@@ -540,6 +540,8 @@
     """Keeps the :py:class:`~niworkflows.utils.spaces.SpatialReferences`
     instance keeping standard and nonstandard spaces."""
     use_aroma = None
+    use_marss = False
+    marss_mb = None
     """Run ICA-:abbr:`AROMA (automatic removal of motion artifacts)`."""
     use_bbr = None
     """Run boundary-based registration for BOLD-to-T1w registration."""
